### PR TITLE
add support for doing a passthrough of requests to preferred backend for unregistered routes

### PR DESCRIPTION
**Pull request by sandeepsukhani**
**What this PR does**:
Adds support for doing a passthrough of requests to the preferred backend for unregistered routes.
This helps with proxying tail requests in Loki which uses WebSockets and are long-living so they need to be passed to a single backend.
I have tested this change already by running it in front of 2 Loki clusters and it has been working fine for more than a week.

**Checklist**
- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/operations/query-tee.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ The following Prometheus API endpoints are supported by `query-tee`:
 - `/api/v1/alerts` (GET)
 - `/api/v1/rules` (GET)
 
+#### PassThrough requests
```
Comment (pracucci) [inline docs/operations/query-tee.md:42]:
```suggestion
#### Pass-through requests
```

File: docs/operations/query-tee.md [hunk 2]
```diff
@@ -39,6 +39,12 @@ The following Prometheus API endpoints are supported by `query-tee`:
...
+
+`query-tee` supports acting as a PassThrough for requests to routes not matching any of the documented `API endpoints above`.
```
Comment (pracucci) [inline docs/operations/query-tee.md:44]:
```suggestion
`query-tee` supports acting as a transparent proxy for requests to routes not matching any of the documented API endpoints above.
```

File: docs/operations/query-tee.md [hunk 3]
```diff
@@ -39,6 +39,12 @@ The following Prometheus API endpoints are supported by `query-tee`:
...
+When enabled, those requests are passed on to just the configured preferred backend.
+To activate this feature it requires setting `-proxy.passthrough-non-registered-routes` flag to `true` and configuring a preferred backend.
```
Comment (pracucci) [inline docs/operations/query-tee.md:46]:
```suggestion
To activate this feature it requires setting `-proxy.passthrough-non-registered-routes=true` flag and configuring a preferred backend.
```

**Comment by sandeepsukhani** [comment]
> Amazing! Very good job: the doc, the tests, the feature (obviously). Well done

Thanks Marco! Your reviews help me improve :)
Also, thanks for taking care of the suggested changes.

**Pull request merged**